internal/providers/terraform/google/compute_disk.go

```diff
@@ -84,10 +53,38 @@ func computeDiskSize(d *schema.ResourceData) *decimal.Decimal {
 	return defaultDiskSize(d.Get("type").String())
 }
 
-func defaultDiskSize(diskType string) *decimal.Decimal {
+func defaultDiskSize(diskType string) float64 {
 	diskType = strings.ToLower(diskType)
 	if diskType == "pd-balanced" || diskType == "pd-ssd" {
-		return decimalPtr(decimal.NewFromInt(100))
+		return 100
+	}
+	return 500
+}
+
+func computeImageDiskSize(d *schema.ResourceData) float64 {
+	if d.Get("disk_size_gb").Exists() {
+		return d.Get("disk_size_gb").Float()
+	}
+
+	if len(d.References("source_disk")) > 0 {
+		return computeDiskSize(d.References("source_disk")[0])
 	}
-	return decimalPtr(decimal.NewFromInt(500))
+
+	if len(d.References("source_image")) > 0 {
+		return computeImageDiskSize(d.References("source_image")[0])
+	}
+
+	if len(d.References("source_snapshot")) > 0 {
+		return computeSnapshotDiskSize(d.References("source_snapshot")[0])
+	}
+
+	return 0
```

Comment:
Not sure if it is okay to treat zero as "empty" value here.

Reply:
I think it makes sense. It looks like we had a nil ptr risk before: https://github.com/infracost/infracost/blob/438c7a01f56ba6065af841573d1e3e3bb1336084/internal/providers/terraform/google/compute_disk.go#L49. So this solves that.